x-pack/legacy/plugins/code/server/distributed/cluster/node_repositories_service.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import util from 'util';
+import { ClusterService, ClusterStateListener } from './cluster_service';
+import { ClusterStateEvent } from './cluster_state_event';
+import { ClusterMembershipService } from './cluster_membership_service';
+import { CloneWorker } from '../../queue';
+import { Repository, RepositoryUri } from '../../../model';
+import { Logger } from '../../log';
+import { RepoState } from '../../../public/actions';
```

Comment:
nit: not your fault, but one last thing is that this is the server end code, and it might make sense to move the `RepoState` to a more general place like `common` or `model`.

Reply:
OK, I will open another pull request to handle this.